pkg/ccl/backupccl/testdata/backup-restore/restore-check-descriptors

```diff
@@ -0,0 +1,61 @@
+# This test checks that restore automatically detects a corrupt restoring
+# descriptor and enters the reverting state (i.e. an infinite OnFailOrCancel
+# loop with exponential back off, as Restore's OnFailOrCancel implementation
+# cannot fix this specific corrupt descriptor problem automatically) and does
+# not publish restoring database with the corrupt descriptor.
+
+new-server name=s1
+----
+
+exec-sql
+CREATE DATABASE testdb;
+CREATE TABLE testdb.parent (k INT PRIMARY KEY, v STRING);
+INSERT INTO testdb.parent (k, v) VALUES (1, 'a');
+CREATE TABLE testdb.child (k INT NOT NULL, FOREIGN KEY (k) REFERENCES testdb.parent (k))
+----
+
+exec-sql
+BACKUP INTO 'nodelocal://1/full_cluster_backup/';
+----
+
+exec-sql
+SET CLUSTER SETTING jobs.debug.pausepoints = 'restore.before_publishing_descriptors';
+----
+
+exec-sql
+ALTER TABLE testdb.child RENAME TO cool_cousin;
+----
+
+restore expect-pausepoint tag=a
+RESTORE DATABASE testdb FROM LATEST IN 'nodelocal://1/full_cluster_backup/' with new_db_name='d2'
+----
+job paused at pausepoint
+
+# Corrupt the child >;-)
```

Comment:
nit: feel free to ignore, but a line that clarifies that it's the restoring descriptor that has been written but not published that we are corrupting would have helped figure out why we have that rename above.

Reply:
Done.